x-pack/legacy/plugins/ml/public/file_datavisualizer/components/import_view/importer/csv_importer.js

```diff
@@ -58,13 +61,38 @@ export class CsvImporter extends Importer {
   }
 }
 
+// parse numeric and boolean fields - treat everything else as a string
+function shouldUseDynamicType(columnNames, mappings, columnNumber) {
+  if (columnNumber >= columnNames.length) {
+    return false;
+  }
+  const columnMapping = mappings[columnNames[columnNumber]];
+  if (columnMapping === undefined || columnMapping.type === undefined) {
+    return false;
+  }
+  switch (columnMapping.type) {
```

Comment:
Are these Elasticsearch types? If so, you should use the values out of the `ES_FIELD_TYPES` enum in `ml/common/constants/field_types.ts`